### PR TITLE
Bump typescript to v4.7.4

### PR DESCRIPTION
**Pull request by wemeetagain**
**Motivation**

Typescript 4.7 includes additional ESM compatibility that should be explored.

**Description**

Bump typescript now, figure out configuration later

### CODE REVIEW

**Comment by dapplion** [comment]
:rocket: 

Since 4.7 supports `node16` and `nodenext` `compilerOptions.module` option, should we change it? I'm not sure what practical implications does it have to us or if we should use it.

https://devblogs.microsoft.com/typescript/announcing-typescript-4-7/?q=1#esm-nodejs

**Comment by wemeetagain** [comment]
maybe we can just merge this as-is and figure out the compiler options later. Will make an issue to track.

**Comment by wemeetagain** [comment]
- https://github.com/ChainSafe/lodestar/issues/4152

**Comment by philknows** [comment]
Anything blocking this upgrade from happening? Should know the priority of this so it doesn't just sit here.

**Comment by wemeetagain** [comment]
Nothing blocking this. Also typescript is up to 4.7.4.
Imo low priority but also like a 1 line change so I'll update this and we can merge it.

File: packages/lodestar/src/chain/blocks/importBlock.ts [hunk 1]
```diff
@@ -187,6 +192,35 @@ export async function importBlock(chain: ImportBlockModules, fullyVerifiedBlock:
     const cp = getCheckpointFromState(checkpointState);
     chain.checkpointStateCache.add(cp, checkpointState);
     pendingEvents.push(ChainEvent.checkpoint, cp, checkpointState);
+
+    // Note: in-lined code from previos handler of ChainEvent.checkpoint
+    chain.logger.verbose("Checkpoint processed", toCheckpointHex(cp));
```
Comment (dapplion) [inline packages/lodestar/src/chain/blocks/importBlock.ts]: What's up with this diff? @wemeetagain

Comment (wemeetagain) [reply packages/lodestar/src/chain/blocks/importBlock.ts]: I think its bc of the force-push to `unstable`

Comment (wemeetagain) [reply packages/lodestar/src/chain/blocks/importBlock.ts]: should be fixed now

**Review by dapplion — APPROVED** [review]

**Pull request merged**